include/multigrid_pclomp/multi_voxel_grid_covariance_omp.h

```diff
@@ -0,0 +1,423 @@
+          */
+        Eigen::Matrix3d
+        getCov () const
+        {
+          return (cov_);
+        }
+
+        /** \brief Get the inverse of the voxel covariance.
+          * \return inverse covariance matrix
+          */
+        Eigen::Matrix3d
+        getInverseCov () const
+        {
+          return (icov_);
+        }
+
+        /** \brief Get the voxel centroid.
+          * \return centroid
+          */
+        Eigen::Vector3d
+        getMean () const
+        {
+          return (mean_);
+        }
+
+        /** \brief Get the eigen vectors of the voxel covariance.
+          * \note Order corresponds with \ref getEvals
+          * \return matrix whose columns contain eigen vectors
+          */
+        Eigen::Matrix3d
+        getEvecs () const
+        {
+          return (evecs_);
+        }
+
+        /** \brief Get the eigen values of the voxel covariance.
+          * \note Order corresponds with \ref getEvecs
+          * \return vector of eigen values
+          */
+        Eigen::Vector3d
+        getEvals () const
+        {
+          return (evals_);
+        }
+
+        /** \brief Get the number of points contained by this voxel.
+          * \return number of points
+          */
+        int
+        getPointCount () const
+        {
+          return (nr_points);
+        }
+
+        /** \brief Number of points contained by voxel */
+        int nr_points;
+
+        /** \brief 3D voxel centroid */
+        Eigen::Vector3d mean_;
+
+        /** \brief Nd voxel centroid
+         * \note Differs from \ref mean_ when color data is used
+         */
+        Eigen::VectorXf centroid;
+
+        /** \brief Voxel covariance matrix */
+        Eigen::Matrix3d cov_;
+
+        /** \brief Inverse of voxel covariance matrix */
+        Eigen::Matrix3d icov_;
+
+        /** \brief Eigen vectors of voxel covariance matrix */
+        Eigen::Matrix3d evecs_;
+
+        /** \brief Eigen values of voxel covariance matrix */
+        Eigen::Vector3d evals_;
+
+      };
+
+      struct LeafID {
+        std::string parent_grid_id;
+        int leaf_index;
+        bool operator < (const LeafID& rhs) const {
+          if (parent_grid_id < rhs.parent_grid_id) {
+            return true;
+          }
+          if (parent_grid_id > rhs.parent_grid_id) {
+            return false;
+          }
+          if (leaf_index < rhs.leaf_index) {
+            return true;
+          }
+          if (leaf_index > rhs.leaf_index) {
+            return false;
+          }
+          return false;
+        }
+      };
+
+      /** \brief Pointer to MultiVoxelGridCovariance leaf structure */
+      typedef Leaf* LeafPtr;
+
+      /** \brief Const pointer to MultiVoxelGridCovariance leaf structure */
+      typedef const Leaf* LeafConstPtr;
+
+      typedef std::map<LeafID, Leaf> Map;
+
+      struct VoxelGridInfo {
+        PointCloud voxel_centroids;
+        Map leaves;
+        std::vector<LeafID> leaf_indices;
```

Comment:
`leaves` includes information on `voxel_centroids` and `leaf_indices`.

Reply:
Thank you, you are right.
I removed the VoxelGridInfo struct itself in the latest code. Please check the latest three commits.
https://github.com/tier4/ndt_omp/pull/19/commits/77a104015afc82db656443aeed781168b2a0265e
https://github.com/tier4/ndt_omp/pull/19/commits/c632fc80b9f9f04851900c4fa85d66fe5bf3717b
https://github.com/tier4/ndt_omp/pull/19/commits/c63423b3fd5d8089a99c649e392d4825c6c1a6d4